cli/src/main/java/org/jboss/as/cli/handlers/trycatch/TryCatchFinallyControlFlow.java

```diff
@@ -97,6 +98,9 @@ boolean isInFinally() {
     }
 
     void moveToCatch() throws CommandLineException {
+        if(catchList == null) {
```

Comment:
You could move the list initialisation in the switch (IN_TRY). Doing so, no need to check for null and you will only initialise the list when needed.